src/openassetio-core/src/utils/formatter.cpp

```diff
@@ -0,0 +1,198 @@
+  }
+  fmt::formatter<openassetio::Context> valueFormatter;
+  return valueFormatter.format(*context, ctx);
+}
+
+auto fmt::formatter<openassetio::ContextConstPtr>::format(
+    const openassetio::ContextConstPtr& context, format_context& ctx) const
+    -> decltype(ctx.out()) {
+  if (context == nullptr) {
+    return fmt::formatter<string_view>::format("null", ctx);
+  }
+  fmt::formatter<openassetio::Context> valueFormatter;
+  return valueFormatter.format(*context, ctx);
+}
+
+auto fmt::formatter<openassetio::Context>::format(const openassetio::Context& context,
+                                                  format_context& ctx) const
+    -> decltype(ctx.out()) {
+  openassetio::Str out = fmt::format("{{'locale': {}, 'managerState': {}}}", context.locale,
+                                     static_cast<void*>(context.managerState.get()));
+
+  return fmt::formatter<string_view>::format(out, ctx);
+}
+
+auto fmt::formatter<std::shared_ptr<openassetio::trait::TraitsData>>::format(
+    const openassetio::trait::TraitsDataPtr& traitsData, format_context& ctx) const
+    -> decltype(ctx.out()) {
+  if (traitsData == nullptr) {
+    return fmt::formatter<string_view>::format("null", ctx);
+  }
+  fmt::formatter<openassetio::trait::TraitsData> valueFormatter;
+  return valueFormatter.format(*traitsData, ctx);
+}
+
+auto fmt::formatter<openassetio::trait::TraitsDataConstPtr>::format(
+    const openassetio::trait::TraitsDataConstPtr& traitsData, format_context& ctx) const
+    -> decltype(ctx.out()) {
+  if (traitsData == nullptr) {
+    return fmt::formatter<string_view>::format("null", ctx);
+  }
+  fmt::formatter<openassetio::trait::TraitsData> valueFormatter;
+  return valueFormatter.format(*traitsData, ctx);
+}
+
+auto fmt::formatter<openassetio::trait::TraitsData>::format(
+    const openassetio::trait::TraitsData& traitsData, format_context& ctx) const
+    -> decltype(ctx.out()) {
+  const auto traitSet = traitsData.traitSet();
+  std::vector<openassetio::Str> traitStrings;
+  traitStrings.reserve(traitSet.size());
+
+  for (const auto& traitId : traitSet) {
+    std::vector<std::string> propertyStrings;
+    const auto keys = traitsData.traitPropertyKeys(traitId);
+    std::transform(std::cbegin(keys), std::cend(keys), std::back_inserter(propertyStrings),
+                   [&traitsData, &traitId](const std::string& propertyKey) {
+                     openassetio::trait::property::Value out;
+                     traitsData.getTraitProperty(&out, traitId, propertyKey);
+                     return fmt::format(R"('{}': {})", propertyKey, out);
+                   });
+
+    // Add the string that makes up the trait, which is the json dict section for the traitId,
+    // followed by all the property key value pairs.
+    traitStrings.push_back(fmt::format("'{}': {{{}}}", traitId, fmt::join(propertyStrings, ", ")));
+  }
+
+  // The idea here being that this is a valid json dict, hence all the extra brace formatting
```

Comment:
[fix] Not JSON dict, Python dict.  The single quotes make it invalid JSON, sadly.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1329/commits/6eb7b04db9a17edd720dd1edf8f6f91bd2a7988c